chia/full_node/weight_proof.py

```diff
@@ -1480,19 +1473,21 @@ def __get_cc_sub_slot(sub_slots: List[SubSlotData], idx, ses: Optional[SubEpochS
     return cc_sub_slot
 
 
-def _get_curr_diff_ssi(constants: ConsensusConstants, idx, summaries):
+def _get_curr_diff_ssi(
+    constants: ConsensusConstants, idx: int, summaries: List[SubEpochSummary]
+) -> Tuple[uint64, uint64]:
     curr_difficulty = constants.DIFFICULTY_STARTING
     curr_ssi = constants.SUB_SLOT_ITERS_STARTING
     for ses in reversed(summaries[0:idx]):
-        if ses.new_sub_slot_iters is not None:
+        if ses.new_sub_slot_iters is not None and ses.new_difficulty is not None:
```

Comment:
is it reasonable to check for this here, or should this be an assert below the if-block instead?

Reply:
i think it would be easier to understand whats going on if we add an assert